code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -159,6 +159,20 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 		return
 	}
 
+	payload.Amount = amount
+	for _, sig := range payload.Signatures {
+		err = ctx.AddMint(&state.Mint{
+			Minter:     gn.ID,
+			ToClientID: sig.ID,
+			Amount:     share,
+			IsBridge:   true,
```

Comment:
do not extend mint as well, since Mint is something common, and it shouldn't know anything about smart contract logic like bridging. you can emit burn event right from here, check https://github.com/0chain/0chain/pull/2230 here is this event already created

Reply:
I will do

Reply:
Re 2230, If it's merged before this PR, I'll remove the definition of the event from here.